### PR TITLE
Code upload optimize

### PR DESCRIPTION
**Pull request by Ic3Sandy**

### CODE REVIEW

File: docker-compose.fvm.yml [hunk 1]
```diff
@@ -23,6 +23,23 @@ services:
...
       - db
 
+  storage:
+    image: node:18-alpine3.14
+    restart: always
+    command: sh -c "npm i && npm run  build && npm run start"
+    stdin_open: true
+    tty: true
+    ports:
+      - 5555:5555
```
Comment (thebevrishot) [inline docker-compose.fvm.yml]: For security reason here, we should not enable public access.

File: service-storage/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import * as fs from 'fs';
+import basicAuth from 'express-basic-auth';
+import { folderUpload, GGStorage } from './storage';
+
+const app: Express = express();
+const port = process.env.PORT;
+app.use(express.json());
+app.use(
+  basicAuth({
+    users: { admin: 'Admin123!' },
```
Comment (thebevrishot) [inline service-storage/index.ts]: this should read from file

File: service-storage/index.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    code,
+  };
+  await fs.promises.writeFile(`${folderUpload}/${contractAddress}.json`, JSON.stringify(jsonObj), 'utf8');
+  const ggStorage = new GGStorage();
+  await ggStorage.uploadFile(contractAddress);
+  res.sendStatus(200);
+});
+
+app.listen(port, () => {
+  console.log(`[server]: Server is running at http://localhost:${port}`);
```
Comment (thebevrishot) [inline service-storage/index.ts]: Not sure if we only run on localhost? so we should not call this other docker containers

File: packages/bitcore-node/src/modules/firocoin/utils.ts [hunk 4]
```diff
@@ -209,3 +210,32 @@ export const decodeLogs = async (logs, contractAddress = '') => {
 export const formatHexAddress = (address: string) => {
   return address.replace('0x', '').toLowerCase();
 };
+
+export const fetchGetContract = async (address: string) => {
+  const response = await fetch(`http://storage:5555/contracts/${address}`, {
+    method: 'get',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: 'Basic ' + Buffer.from('admin:Admin123!').toString('base64'),
```
Comment (thebevrishot) [inline packages/bitcore-node/src/modules/firocoin/utils.ts]: Please read from file

File: packages/bitcore-node/src/modules/firocoin/utils.ts [hunk 5]
```diff
@@ -209,3 +210,32 @@ export const decodeLogs = async (logs, contractAddress = '') => {
 export const formatHexAddress = (address: string) => {
   return address.replace('0x', '').toLowerCase();
 };
+
+export const fetchGetContract = async (address: string) => {
+  const response = await fetch(`http://storage:5555/contracts/${address}`, {
```
Comment (thebevrishot) [inline packages/bitcore-node/src/modules/firocoin/utils.ts]: This should be a configuration

**Review by thebevrishot — CHANGES_REQUESTED** [review]
We should not store files in local bitcore

File: docker-compose.fvm.yml (hunk 1 above)
Comment (Ic3Sandy) [reply docker-compose.fvm.yml]: ok kub

File: service-storage/index.ts (hunk 2 above)
Comment (Ic3Sandy) [reply service-storage/index.ts]: ok krub

File: packages/bitcore-node/src/modules/firocoin/utils.ts (hunk 4 above)
Comment (Ic3Sandy) [reply packages/bitcore-node/src/modules/firocoin/utils.ts]: ok kub

File: packages/bitcore-node/src/modules/firocoin/utils.ts (hunk 5 above)
Comment (Ic3Sandy) [reply packages/bitcore-node/src/modules/firocoin/utils.ts]: ok kub

File: service-storage/index.ts (hunk 3 above)
Comment (Ic3Sandy) [reply service-storage/index.ts]: I will fix this kub

**Pull request merged**